src/store/modules/orders/actions.ts

```diff
@@ -0,0 +1,192 @@
+import { OrderService } from "@/services/OrderService";
+import { ActionTree } from 'vuex'
+import RootState from '@/store/RootState'
+import OrdersState from './OrdersState'
+import * as types from './mutation-types'
+import { hasError , showToast } from "@/utils";
+import { translate } from "@/i18n";
+import emitter from '@/event-bus'
+import router from "@/router";
+
+const actions: ActionTree<OrdersState , RootState> ={
+  async getOrder ({ commit, state }, payload) {
+    // Show loader only when new query and not the infinite scroll
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+
+    try {
+      resp = await OrderService.getOrders(payload)
+      if (resp.status === 200 && resp.data.count > 0 && !hasError(resp)) {
+        let orders = resp.data.docs;
+        const ordersCount = resp.data.count;
+        if(payload.viewIndex && payload.viewIndex > 0) orders = state.orders.list.concat(orders)
+        commit(types.OPEN_ORDERS_INITIAL, {orders: orders , ordersCount: ordersCount})
+        if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+      } else {
+        showToast(translate("Orders Not Found"))
+      }
+      if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  updateCurrentOrder ({ commit }, payload) {
+    commit(types.ORDERS_CURRENT_UPDATED, { order: payload.order })
+  },
+
+  async getPackedOrders ({ commit, state }, payload) {
+    // Show loader only when new query and not the infinite scroll
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+
+    try {
+      resp = await OrderService.getPackedOrders(payload)
+      if (resp.status === 200 && resp.data.count > 0 && !hasError(resp)) {
+        let packedOrders = resp.data.docs;
+        const total = resp.data.count;
+        if(payload.viewIndex && payload.viewIndex > 0) packedOrders = state.packedOrders.list.concat(packedOrders)
+        commit(types.ORDERS_PACKED_INITIAL, { packedOrders, total })
+        if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+      } else {
+        showToast(translate("Orders Not Found"))
+      }
+      if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  async deliverShipment ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+
+    const params = {
+      shipmentId: order.shipmentId,
+      statusId: 'SHIPMENT_SHIPPED'
+    }
+
+    let resp;
+
+    try {
+      resp = await OrderService.updateShipment(params)
+      if (resp.status === 200 && !hasError(resp)) {
+        showToast(translate(`Order delivered to ${order.customerName}`))
```

Comment:
Translation won't work here as we are passing dynamic value. We need to pass customer name separately to translate method

Reply:
Changed the definition for translate method and updated the code.